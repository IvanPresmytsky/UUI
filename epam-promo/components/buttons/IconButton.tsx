import { createSkinComponent } from '@epam/uui-core';
import { EpamPrimaryColor } from '../types';
import * as uui from '@epam/uui';

export type IconColor = EpamPrimaryColor | 'gray30' | 'gray50' | 'gray60';

export interface IconButtonMods {
    /**
     * @default 'gray60'
     */
    color?: IconColor;
}

export type IconButtonProps = uui.IconButtonCoreProps & IconButtonMods;

export const IconButton = createSkinComponent<uui.IconButtonProps, IconButtonProps>(
    uui.IconButton,
    (props) => ({ color: props.color ?? 'gray60' }),
);
