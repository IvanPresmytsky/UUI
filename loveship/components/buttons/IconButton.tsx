import * as uui from '@epam/uui';
import { createSkinComponent, devLogger } from '@epam/uui-core';
import { EpamAdditionalColor, EpamPrimaryColor } from '../types';

type IconButtonMods = {
    /**
     * Defines component color.
     * @default 'night600'
     */
    color?: EpamPrimaryColor | EpamAdditionalColor | 'white' | 'night200' | 'night300' | 'night400' | 'night500' | 'night600';
};

/** Represents the properties of a IconButton component. */
export type IconButtonProps = uui.IconButtonCoreProps & IconButtonMods;

export const IconButton = createSkinComponent<uui.IconButtonProps, IconButtonProps>(
    uui.IconButton,
    (props) => {
        if (__DEV__) {
            devLogger.warnAboutDeprecatedPropValue<IconButtonProps, 'color'>({
                component: 'IconButton',
                propName: 'color',
                propValue: props.color,
                condition: () => ['night200', 'night300', 'night400'].indexOf(props.color) !== -1,
            });
        }
        return {
            color: props.color ?? 'night600',
        };
    },
);
