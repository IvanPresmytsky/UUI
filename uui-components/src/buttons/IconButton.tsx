import * as React from 'react';
import { ButtonBaseProps, ButtonBase } from './ButtonBase';
import { IconContainer } from '../layout';
import css from './Button.module.scss';

export type IconButtonBaseProps<T> = ButtonBaseProps<T> & {};

export class IconButton<T> extends ButtonBase<IconButtonBaseProps<T>> {
    constructor(props: IconButtonBaseProps<T>) {
        super(props);
    }

    getClassName() {
        return [css.container];
    }

    getChildren() {
        return <IconContainer key="icon" icon={ this.props.icon } />;
    }
}
