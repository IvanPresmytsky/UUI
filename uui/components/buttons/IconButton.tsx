import * as uuiComponents from '@epam/uui-components';
import { withMods } from '@epam/uui-core';
import css from './IconButton.module.scss';
import { systemIcons } from '../../icons/icons';

interface IconButtonMods {
    /**
     * Defines component color.
     * @default 'neutral'
     */
    color?: 'info' | 'success' | 'warning' | 'error' | 'secondary' | 'neutral';
    /**
     * Defines component size.
     * @default '18'
     */
    size?: '18' | '24' | '30' | '36';
}

const DEFAULT_SIZE: IconButtonMods['size'] = '18';

/** Represents the Core properties of the IconButton component. */
export type IconButtonCoreProps = uuiComponents.IconButtonProps;

/** Represents the properties of the IconButton component. */
export type IconButtonProps = Omit<IconButtonCoreProps, 'size'> & IconButtonMods;

function applyIconButtonMods(mods: IconButtonProps & IconButtonMods) {
    return ['uui-icon_button', `uui-color-${mods.color || 'neutral'}`, css.root];
}

export const IconButton = withMods<Omit<IconButtonCoreProps, 'size'>, IconButtonMods>(
    uuiComponents.IconButton,
    applyIconButtonMods,
    (props) => {
        const size = props.size ? Number(props.size) : Number(DEFAULT_SIZE);
        return {
            dropdownIcon: props.dropdownIcon || systemIcons.foldingArrow,
            size,
        };
    },
);
