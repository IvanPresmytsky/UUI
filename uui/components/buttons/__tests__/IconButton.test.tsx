import React from 'react';
import { IconButton } from '../IconButton';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';
import { ReactComponent as CalendarIcon } from '../../../icons/calendar-24.svg';

describe('IconButton', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<IconButton />);
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <IconButton
                color="info"
                onClick={ jest.fn }
                icon={ CalendarIcon }
                isDisabled={ false }
            />,
        );
        expect(tree).toMatchSnapshot();
    });
});
