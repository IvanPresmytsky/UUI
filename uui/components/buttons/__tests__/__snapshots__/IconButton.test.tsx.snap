// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should be rendered correctly 1`] = `
<button
  className="uui-button-box uui-enabled container uui-icon_button uui-color-neutral root"
  tabIndex={-1}
  type="button"
>
  <div
    className="container uui-icon uui-enabled"
    style={{}}
  />
</button>
`;

exports[`IconButton should be rendered correctly with props 1`] = `
<button
  aria-disabled={false}
  className="uui-button-box uui-enabled -clickable container uui-icon_button uui-color-info root"
  onClick={[Function]}
  tabIndex={0}
  type="button"
>
  <div
    className="container uui-icon uui-enabled"
    style={{}}
  >
    <svg
      className=""
    />
  </div>
</button>
`;
